packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/create-reservation/components/create-reservation-form/create-reservation-form.tsx

```diff
@@ -0,0 +1,281 @@
+import * as zod from "zod"
+
+import { Button, Heading, Input, Text, Textarea } from "@medusajs/ui"
+import {
+  RouteFocusModal,
+  useRouteModal,
+} from "../../../../../../components/route-modal"
+import { useInventoryItems } from "../../../../../../hooks/api/inventory"
+
+import { Combobox } from "../../../../../../components/common/combobox"
+import { Form } from "../../../../../../components/common/form"
+import { InventoryItemRes } from "../../../../../../types/api-responses"
+import { InventoryNext } from "@medusajs/types"
+import React from "react"
+import { useForm } from "react-hook-form"
+import { useStockLocations } from "../../../../../../hooks/api/stock-locations"
+import { useTranslation } from "react-i18next"
+import { zodResolver } from "@hookform/resolvers/zod"
+import { useCreateReservationItem } from "../../../../../../hooks/api/reservations"
+
+export const CreateReservationSchema = zod.object({
+  inventory_item_id: zod.string().min(1),
+  location_id: zod.string().min(1),
+  quantity: zod.number().min(1),
+  description: zod.string().optional(),
+})
+
+const AttributeGridRow = ({
+  title,
+  value,
+}: {
+  title: string
+  value: string | number
+}) => {
+  return (
+    <div className="grid grid-cols-2 divide-x">
+      <Text className="px-2 py-1.5" size="small" leading="compact">
+        {title}
+      </Text>
+      <Text className="px-2 py-1.5" size="small" leading="compact">
+        {value}
+      </Text>
+    </div>
+  )
+}
+
+export const CreateReservationForm = () => {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+  const [inventorySearch, setInventorySearch] = React.useState<string | null>(
+    null
+  )
+
+  const form = useForm<zod.infer<typeof CreateReservationSchema>>({
+    defaultValues: {
+      inventory_item_id: "",
+      location_id: "",
+      quantity: 0,
+      description: "",
+    },
+    resolver: zodResolver(CreateReservationSchema),
+  })
+
+  const { inventory_items } = useInventoryItems(
+    {
+      q: inventorySearch,
+    },
+    {
+      enabled: !!inventorySearch,
+    }
+  )
+
+  const inventoryItemId = form.watch("inventory_item_id")
+  const selectedInventroyItem = inventory_items?.find(
+    (it) => it.id === inventoryItemId
+  ) as InventoryItemRes["inventory_item"] | undefined
+
+  const locationId = form.watch("location_id")
+  const selectedLocationLevel = selectedInventroyItem?.location_levels?.find(
+    (it) => it.location_id === locationId
+  )
+
+  const quantity = form.watch("quantity")
+
+  const { stock_locations } = useStockLocations(
+    {
+      id:
+        selectedInventroyItem?.location_levels?.map(
+          (level: InventoryNext.InventoryLevelDTO) => level.location_id
+        ) ?? [],
+    },
+    {
+      enabled: !!selectedInventroyItem,
```

Comment:
suggestion: either disable location select until the item is selected or show a tooltip for location saying that inventory needs to be selected first